lib/posix/CMakeLists.txt

```diff
@@ -26,13 +27,12 @@ zephyr_library_sources_ifdef(CONFIG_POSIX_FS fs.c)
 zephyr_library_sources_ifdef(CONFIG_EVENTFD eventfd.c)
 add_subdirectory_ifdef(CONFIG_GETOPT getopt)
 
-if(NOT (CONFIG_BOARD_NATIVE_POSIX OR CONFIG_BOARD_NATIVE_POSIX_64BIT))
-zephyr_library_sources(nanosleep.c)
-endif()
-
 zephyr_library_include_directories(
   ${ZEPHYR_BASE}/kernel/include
   ${ARCH_DIR}/${ARCH}/include
 )
 
-zephyr_library_link_libraries(posix_subsys)
+if (CONFIG_POSIX_API OR CONFIG_PTHREAD_IPC OR CONFIG_POSIX_CLOCK OR
+    CONFIG_POSIX_MQUEUE OR CONFIG_POSIX_FS OR CONFIG_EVENTFD OR CONFIG_GETOPT)
+  zephyr_library_link_libraries(posix_subsys)
+endif()
```

Comment:
well, this is messy ... ideally, there would be a top-level POSIX subsystem symbol (e.g. `CONFIG_POSIX`), but that is not the case at the moment.

another option would be to just set `zephyr_library_property(ALLOW_EMPTY TRUE)`.